src/ClearSale/XmlEntity/SendOrder/Address.php

```diff
@@ -57,13 +57,11 @@ public function getNumber()
 
     public function setNumber($number)
     {
-        $number = preg_replace('/[^0-9]/', '', $number);
-
         if (empty($number)) {
             throw new InvalidArgumentException('Number is empty!');
         }
 
-        $this->number = $number;
+        $this->number = "$number";
```

Comment:
Talvez fazer um cast explícito com o (string) ao invés de encapsular a variável com aspas duplas seja melhor, não acha?
$this->number = (string) $number;
Fonte: http://www.php.net/manual/en/language.types.string.php#language.types.string.casting